pkg/storage/kubernetes/ipam.go

```diff
@@ -116,7 +116,7 @@ func (i *KubernetesIPAM) GetIPPool(ctx context.Context, ipRange string) (storage
 	return &KubernetesIPPool{i.client, i.containerID, firstIP, pool}, nil
 }
 
-func NormalizeRange(ipRange string) string {
+func NormalizeRange(ipRange string, networkName string) string {
```

Comment:
hm, we need to come up with a better way to do this: this function normalized a range before, but actually, what this did was generate the name of the `IPPool`. That's made more obvious now.

I propose we make this clearer (would appreciate a better name for the `GlobalIPPoolName` though):
```golang
func GlobalIPPoolName(ipRange string) string {
    return normalizedRange(ipRange)
}

func IPPoolName(networkName string, ipRange string) string {
    return fmt.Sprintf("%s-%s", networkName, normalizedRange(ipRange))
}

func normalizedRange(ipRange string) string {
	if ipRange[len(ipRange)-1] == ':' {
		ipRange = ipRange + "0"
	}
	normalized := strings.ReplaceAll(ipRange, ":", "-")
	normalized = strings.ReplaceAll(normalized, "/", "-")
	return normalized
}
```

And then let the caller choose which to use - the production code caller can stay as you have (well, it'll need to check which of the 2 funcs to call ...), but the e2e tests tests will IMO be a lot easier to follow.

Reply:
✅